adapters/akka-http/src/main/scala/caliban/AkkaHttpAdapter.scala

```diff
@@ -251,18 +258,18 @@ object AkkaHttpAdapter {
   /**
    * ContextWrapper provides a way to pass context from http request into Caliban's query handling.
    */
-  trait ContextWrapper { self =>
-    def apply[R](ctx: RequestContext)(e: URIO[R, HttpResponse]): URIO[R, HttpResponse]
+  trait ContextWrapper[-R, A] { self =>
+    def apply[R1 <: R](ctx: RequestContext)(e: URIO[R1, A]): URIO[R1, A]
 
-    def |+|(that: ContextWrapper): ContextWrapper = new ContextWrapper {
-      override def apply[R](ctx: RequestContext)(e: URIO[R, HttpResponse]): URIO[R, HttpResponse] =
+    def |+|[R1 <: R](that: ContextWrapper[R1, A]): ContextWrapper[R1, A] = new ContextWrapper[R1, A] {
+      override def apply[R2 <: R1](ctx: RequestContext)(e: URIO[R2, A]): URIO[R2, A] =
         that.apply(ctx)(self.apply(ctx)(e))
     }
   }
 
   object ContextWrapper {
-    def empty: ContextWrapper = new ContextWrapper {
-      override def apply[R](ctx: RequestContext)(effect: URIO[R, HttpResponse]): URIO[R, HttpResponse] =
+    def empty[A]: ContextWrapper[Any, A] = new ContextWrapper[Any, A] {
```

Comment:
If you make `ContextWrapper` covariant in `A`, you can remove the `[A]` type parameter and fix it to `Nothing`.

Reply:
Don't think we can do this since the appender function's `that` parameter is in a contravariant position

Reply:
You can, you just have to use the same trick we use for `R`:
```scala
  trait ContextWrapper[-R, +A] { self =>
    def apply[R1 <: R, A1 >: A](ctx: RequestContext)(e: URIO[R1, A1]): URIO[R1, A1]
  }
```

Reply:
Many thanks for the advice.  If I change this I am struggling to find the right combination to be able to take in `e` effect parameter in the `apply` definition.  For example, something like below does not work

```scala
    def |+|[R1 <: R, A1 >: A](that: ContextWrapper[R1, A1]): ContextWrapper[R1, A1] = new ContextWrapper[R1, A1] {
      override def apply[R2 <: R1, A2 >: A1](ctx: RequestContext)(e: URIO[R2, A2]): URIO[R2, A2] =
        that.apply(ctx)(self.apply(ctx)(e))
    }
```

Reply:
Does it help to specify the types explicitly? `apply[R2, A2]`

Reply:
Yes, thanks, that did the trick!  I have pushed the suggested changes.